sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClientBuilder.java

```diff
@@ -349,20 +325,20 @@ public TableServiceClientBuilder serviceVersion(TablesServiceVersion version) {
     }
 
     /**
-     * Sets the request retry options for all the requests made through the client.
+     * Sets the request retry policy for all the requests made through the client.
      *
-     * @param retryOptions {@link RequestRetryOptions}.
+     * @param retryPolicy {@link RetryPolicy}.
      *
      * @return The updated {@link TableServiceClientBuilder}.
      *
-     * @throws NullPointerException if {@code retryOptions} is {@code null}.
+     * @throws NullPointerException if {@code retryPolicy} is {@code null}.
      */
-    public TableServiceClientBuilder retryOptions(RequestRetryOptions retryOptions) {
-        if (retryOptions == null) {
-            throw logger.logExceptionAsError(new NullPointerException("'retryOptions' cannot be null."));
+    public TableServiceClientBuilder retryOptions(RetryPolicy retryPolicy) {
```

Comment:
The method name should be `retryPolicy` and should allow null values.

Reply:
Good catch! I will make the appropriate changes before merging.